internal/k8s/controllers/http_route_controller.go

```diff
@@ -53,5 +61,123 @@ func (r *HTTPRouteReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 func (r *HTTPRouteReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&gateway.HTTPRoute{}).
+		Watches(
+			&source.Kind{Type: &gateway.ReferencePolicy{}},
+			handler.EnqueueRequestsFromMapFunc(r.referencePolicyToRouteRequests),
+		).
```

Comment:
Do we need this same watch for `TCPRoutes`?

Reply:
I see https://github.com/hashicorp/consul-api-gateway/pull/156/files#diff-764889fcb4f05bae8ffc1396db1a0fc5f2c7d088ef9498edb3b546084d2fa936R166 and am wondering if we're intentionally only dealing with `HTTPRoutes` for this release or will need a followup PR before we release v0.2.0

Reply:
I had been intending to limit the scope of this to `HTTPRoute` for the initial implementation - there are several TODOs throughout for refactoring/generalizing to other route types but I didn't want to bloat this scope initially.

Reply:
Do you know if we need to generalize ahead of the release, or is that a thing we can do for 0.3.0?

Reply:
Discussed in Slack that we want to handle for TCPRoutes as well

Reply:
Discussed that we want to get this in for the same release as this logic for HTTPRoute, but planning the implement as a separate PR following this one.